packages/mobile/src/account/Persona.tsx

```diff
@@ -0,0 +1,124 @@
+
+  const accountAddress = useSelector(accountAddressSelector)
+
+  const dispatch = useDispatch()
+
+  const templateIdResponse = useAsync(async () => readOnceFromFirebase('persona/templateId'), [])
+  const templateId = templateIdResponse.result
+
+  const launchPersonaInquiry = useCallback(() => {
+    if (typeof templateId !== 'string') {
+      Logger.error(TAG, `Attempted to initiate Persona with invalid templateId: ${templateId}`)
+      return
+    }
+
+    if (!accountAddress) {
+      Logger.error(TAG, "Can't render Persona because accountAddress is null")
+      return
+    }
+
+    Inquiry.fromTemplate(templateId)
+      .referenceId(accountAddress)
+      .environment(networkConfig.personaEnvironment)
+      .onSuccess((inquiryId: string, attributes: InquiryAttributes) => {
+        Logger.info(
+          TAG,
+          `Inquiry completed for ${inquiryId} with attributes: ${JSON.stringify(attributes)}`
+        )
+        // TODO [Lisa]: Add event handling for KYC approval when Persona component is integrated
+      })
+      .onCancelled(() => {
+        Logger.info(TAG, 'Inquiry is canceled by the user.')
+      })
+      .onError((error: Error) => {
+        Logger.error(TAG, `Error: ${error.message}`)
+      })
+      .build()
+      .start()
+  }, [templateId])
+
+  const getPrivateKey = async (): Promise<string> => {
+    const mnemonic = await getStoredMnemonic(accountAddress)
+    if (!mnemonic) {
+      throw new Error('Unable to fetch mnemonic from the store')
+    }
+    const keys = await generateKeys(mnemonic)
+    return keys?.privateKey
+  }
+
+  useAsync(async () => {
+    if (!personaAccountCreated) {
+      if (!accountAddress) {
+        Logger.error(TAG, "Can't render Persona because accountAddress is null")
+        return
+      }
+      const privateKey = await getPrivateKey()
+      const signature = signMessage(
+        `post /account/create ${JSON.stringify({ accountAddress })}`,
+        privateKey,
+        accountAddress
+      )
+      const serializedSignature = serializeSignature(signature)
+
+      const response = await fetch(`${networkConfig.inhouseLiquditiyUrl}/persona/account/create`, {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+          authorization: `Valora ${accountAddress}:${serializedSignature}`,
+        },
+        body: JSON.stringify({ accountAddress }),
+      })
+
+      if (response.status === 201 || response.status === 409) {
+        setPersonaAccountCreated(true)
+      } else {
+        dispatch(showError(ErrorMessages.PERSONA_ACCOUNT_ENDPOINT_FAIL))
+      }
+    }
+  }, [])
+
+  return (
+    <Button
+      onPress={launchPersonaInquiry}
+      text={t('raiseLimitBegin')}
+      type={BtnTypes.PRIMARY}
+      size={BtnSizes.FULL}
+      testID="PersonaButton"
+      disabled={
+        !personaAccountCreated
+        // ONLY FOR TESTING PURPOSES
+        // || !templateId
```

Comment:
I commented this out temporarily just to get a working test. All affects from hooks happen after the initial render, so we need to be able to wait for some UI change to verify things like `expect(mockFetch).toHaveBeenCalledTimes(1)`.

One key principle of react-testing-library is that you should keep most of the tests focused on the visual changes the user experiences, in this case the button becomes enabled / disabled based on results from Persona and IHL. In order to truly test this we need to also be able to mock / control `templateId` since it blocks us from testing whether or not the button becomes enabled /disabled.

Next step should be mocking `templateId` from firebase so that you can test this button more effectively

Reply:
oh sweet, just mocked templateId! ty for looking into it